libusb/os/darwin_usb.h

```diff
@@ -71,54 +71,37 @@ typedef IOUSBInterfaceInterface220 **usb_interface_t;
 
 /* IOUSBDeviceInterface */
 
-/* New in OS 10.9.0. */
-#if defined (kIOUSBDeviceInterfaceID650)
-
-#define usb_device_t    IOUSBDeviceInterface650
-#define DeviceInterfaceID kIOUSBDeviceInterfaceID650
-#define DeviceVersion 650
-
-/* New in OS 10.7.3 but can't test deployment target to that granularity, so round up. */
-#elif defined (kIOUSBDeviceInterfaceID500)
-
-#define usb_device_t    IOUSBDeviceInterface500
-#define DeviceInterfaceID kIOUSBDeviceInterfaceID500
-#define DeviceVersion 500
-
-/* New in OS 10.5.4 but can't test deployment target to that granularity, so round up. */
-#elif defined (kIOUSBDeviceInterfaceID320)
-
-#define usb_device_t    IOUSBDeviceInterface320
-#define DeviceInterfaceID kIOUSBDeviceInterfaceID320
-#define DeviceVersion 320
-
-/* New in OS 10.5.0. */
-#elif defined (kIOUSBDeviceInterfaceID300)
-
-#define usb_device_t    IOUSBDeviceInterface300
-#define DeviceInterfaceID kIOUSBDeviceInterfaceID300
-#define DeviceVersion 300
-
-/* New in OS 10.4.5 (or 10.4.6?) but can't test deployment target to that granularity, so round up. */
-#elif defined (kIOUSBDeviceInterfaceID245)
-
-#define usb_device_t    IOUSBDeviceInterface245
-#define DeviceInterfaceID kIOUSBDeviceInterfaceID245
-#define DeviceVersion 245
-
-/* New in OS 10.2.3 but can't test deployment target to that granularity, so round up. */
-#elif defined (kIOUSBDeviceInterfaceID197)
-
-#define usb_device_t    IOUSBDeviceInterface197
-#define DeviceInterfaceID kIOUSBDeviceInterfaceID197
-#define DeviceVersion 197
-
+#if defined(kIOUSBDeviceInterfaceID650)
+#define MAX_DEVICE_VERSION 650
+#elif defined(kIOUSBDeviceInterfaceID500)
+#define	MAX_DEVICE_VERSION 500
+#elif defined(kIOUSBDeviceInterfaceID320)
+#define MAX_DEVICE_VERSION 320
+#elif defined(kIOUSBDeviceInterfaceID300)
+#define MAX_DEVICE_VERSION 300
+#elif defined(kIOUSBDeviceInterfaceID245)
+#define MAX_DEVICE_VERSION 245
 #else
+#define	MAX_DEVICE_VERSION 197
+#endif
 
-#error "IOUSBFamily is too old. Please upgrade your SDK and/or deployment target"
-
+#if (MAC_OS_X_VERSION_MIN_REQUIRED >= 100900)
+typedef IOUSBDeviceInterface650 **usb_device_t;
+#elif (MAC_OS_X_VERSION_MIN_REQUIRED >= 100700)
+typedef IOUSBDeviceInterface500 **usb_device_t;
+#elif (MAC_OS_X_VERSION_MIN_REQUIRED >= 100503)
```

Comment:
The old comment says "New in OS 10.5.4". I checked the macOS SDK and it says the same. So this should be `>= 100504` I think.

Reply:
But wait, also, like the old comment said, back then `MAC_OS_X_VERSION_MIN_REQUIRED` did not have such granularity...

Reply:
Ok, I remember now. Before 10.10 the version was only xxyy major + minor without a patch level. Let me fix this.

Reply:
See https://github.com/JamieBerghmans/Cylinder-iOS13/blob/master/deps/iPhoneOS13.2.sdk/usr/include/AvailabilityMacros.h for reference.

Reply:
Thanks for catching that. This code went through a bunch of revisions and lost that min required can't catch patch releases in 10.9 and earlier.

Reply:
Actually, this is not as useful as it once was. Will just define it to the minimum version. We can bump that up if needed at some point.